lib/components/map_widget.dart

```diff
@@ -182,6 +182,23 @@ class MapWidgetState extends State<MapWidget> {
     List<LatLng> endingPointPath =
         route.path.sublist(route.path.length - 25, route.path.length);
 
+    double startPathLength = 0;
+    double endPathLength = 0;
+
+    for(int i = 0; i < 24; i++){
```

Comment:
this loop never does not produce any results because the length of start and end path with 2 is just to long!